src/wp-includes/nav-menu-template.php

```diff
@@ -199,7 +199,8 @@ function wp_nav_menu( $args = array() ) {
 	$menu_items_with_children = array();
 	foreach ( (array) $menu_items as $menu_item ) {
 		// Fix invalid `menu_item_parent`. See: https://core.trac.wordpress.org/ticket/56926.
-		if ( (int) $menu_item->ID === (int) $menu_item->menu_item_parent ) {
+		// Compare as strings. Plugins may change the ID to string.
+		if ( "$menu_item->ID" === "$menu_item->menu_item_parent" ) {
```

Comment:
```suggestion
		if ( (string) $menu_item->ID === (string) $menu_item->menu_item_parent ) {
```

Opinionated: this seems clearer to me than quotes -- the effect is the same.

Reply:
> this seems clearer to me than quotes

Yep, it is. Casting to string was the first thing I did too :) 

However it changes/casts both `$menu_item->ID` and `$menu_item->menu_item_parent` to strings which might perhaps introduce a bug somewhere else? Both of these are usually numeric (post IDs for the current post and parent post). 

Another possible thing would be to make new vars for these, although seems perhaps too wordy:
```
$menu_item_id     = strval( $menu_item->ID );
$menu_item_parent = strval( $menu_item->menu_item_parent );

if ( $menu_item_id === $menu_item_parent ) {
.....
```

Reply:
> However it changes/casts both `$menu_item->ID` and `$menu_item->menu_item_parent` to strings which might perhaps introduce a bug somewhere else? Both of these are usually numeric (post IDs for the current post and parent post).

Hmm, I wasn't seeing that in my reduced case. Probably because I missed something by reducing it :) 

Let's keep it as is and include an explanation in the comment.

```php
/*
 * Compare as strings as plugins may change the ID to string.
 * To avoid modifying the object, these items are quoted rather than cast to (string).
 */
```

Or something like that anyway.

Reply:
> include an explanation in the comment.

Yes, sounds good.

Also... Re-reading this today thinking this is the most readable and self-documenting code:
```if ( strval( $menu_item->ID ) === strval( $menu_item->menu_item_parent ) )```

`strval()` is equivalent to using double-quotes, and is used at a few other places in core.